r1cs/src/linear_combination.rs

```diff
@@ -91,7 +91,11 @@ impl<F: Field> LinearCombination<F> {
                     found_index += 1;
                 }
             }
-            Err(found_index)
+            if self.0.get(found_index).map(|x| &x.0 == search_var).unwrap_or_default() {
+                Ok(found_index)
+            } else {
+                Err(found_index)
+            }
```

Comment:
Hmm I think this is better resolved by just removing the special case for linear combinations of size smaller than 6; that way the behaviour won't deviate again in the future. I'm sure the rust compiler has special cases for linear scans already.

Reply:
FWIW, it doesn't have any special cases there (I looked to debug this). The logic is tested -- we can keep it if we want it. I assume there was a performance advantage to be had, so I don't see a pressing need to remove it.

Reply:
In that case, maybe it's better to rewrite the logic along the lines of:
```rust
for (i, (var, _)) in self.0.iter().enumerate() {
	use core::ord::Ordering::*;
	match var.cmp(search_var) {
		Equal => return Ok(i),
		Greater => return Err(i),
		_ => (),
	}
}
return Err(self.0.len());
```

Reply:
(It's also not clear to me whether this is actually faster in practice or not; maybe we can add a benchmark.)

Reply:
Lets not fix what isn't broken unless there is a specific reason to. Is there any point to this?

Reply:
Sure, no problem. My rationale was just that it's not clear why the code was wrong earlier, whereas with the new for loop it's clearer what the behaviour is (at least IMO). But that's subjective.